Port several updates over from Registrar

kdmccormick opened the pull request:
https://openedx.atlassian.net/browse/EDUCATOR-4044
@edx/masters-neem 
@edx/testeng 

Registrar was cut from this repo, but over time we made several general changes that we believe would be beneficial to codify in the IDA cookiecutter:
 * New-style MIDDLEWARE declaration
 * pytest instead of nose
 * Import sorting with isort
 * pip-sync instead of pip install
 * Constrain Django to >=1.11.23
 * Dynamic Makefile helptext generation
 * Separate Makefile rules for piptools, pylint, et al
 * Remove redundant pii_check (validate checks it already)
 * Importing url patterns directly instead of using string include
 * codecov.yml
 * Hide unhelpful depr warnings
 * validation.in = test.txt + quality.txt

Additional changes:
 * Add make upgrade target to root
 * Run make upgrade
 * Remove .rst warning from make test
 * Change openedx.yaml files to reference people
 * Remove useless pylint suppressions
 * Update requirements README section
 * Remove redundant oauth_urlpatterns

Note: I also opened https://github.com/edx/jenkins-job-dsl/pull/769

kdmccormick committed:
Port several updates over from Registrar

Registrar was cut from this repo, but over time we made
several general changes that we believe would be beneficial
to codify in the IDA cookiecutter:
 * New-style MIDDLEWARE declaration
 * pytest instead of nose
 * Import sorting with isort
 * pip-sync instead of pip install
 * Constrain Django to >=1.11.23
 * Dynamic Makefile helptext generation
 * Separate Makefile rules for piptools, pylint, et al
 * Remove redundant pii_check (validate checks it already)
 * Importing url patterns directly instead of using string include
 * codecov.yml
 * Hide unhelpful depr warnings
 * validation.in = test.txt + quality.txt

Additional changes:
 * Add make upgrade target to root
 * Run make upgrade
 * Remove .rst warning from make test
 * Change openedx.yaml files to reference people
 * Remove useless pylint suppressions
 * Update requirements README section
 * Remove redundant oauth_urlpatterns

EDUCATOR-4044

kdmccormick commented on openedx.yaml:
```diff
@@ -7,4 +7,7 @@ oeps:
+owner: jmbowman
+supporting_teams:
+  - testeng
+
```
@jmbowman Can you confirm that this change is OK?

kdmccormick replied:
(confirmed through Slack)

kdmccormick commented on {{cookiecutter.repo_name}}/.travis.yml:
```diff
@@ -8,11 +8,10 @@ before_install:
 script:
     - make validate_translations
     - make validate
-    - make pii_check
```
Note for reviewers: PII checking isn't removed; it's just been rolled into `make validate`

iloveagent57 commented on README.rst:
```diff
@@ -63,7 +63,15 @@ You will be prompted for a few basic details (described below). These will be us
+Failure to do so could open your IDA to bugs, security vulnerabilities,
+and other issues.
+
+It is recommended to follow the instructions in
```
👍

iloveagent57 reviewed (approved):
Looks great, nice work.

kdmccormick committed:
Remove version pins from cookie-cutter

This commit removes the .txt files from the
cookie-cutter folder, and instead asks the user
to run 'make upgrade' to generate them after cutting
their new IDA. This way, we do not need to worry about
the version pins from going out-of-date in this
repository.

jansenk commented on {{cookiecutter.repo_name}}/Makefile:
```diff
@@ -1,121 +1,135 @@
+# Generates a help message. Borrowed from https://github.com/pydanny/cookiecutter-djangopackage.
+help: ## display this help message
+	@echo "Please use \`make <target>\` where <target> is one of"
+	@perl -nle'print $& if m{^[\.a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
```
Could you explain what this does?

jmbowman replied:
This displays a colorized help message of available `make` targets, kind of like this:

```
Please use `make <target>' where <target> is one of
  clean                     archive and delete most git-ignored files
  detect_changed_source_translations check if translation files are up-to-date
  docs                      build the developer documentation for this repository
  ...
```

Ned came up with a slightly cleaner version that doesn't depend on perl, we should probably switch to that in both cookiecutters: https://github.com/edx/edx-platform/blob/master/Makefile#L9

jansenk replied:
I can't really make out how it's doing that; is it reading this file and getting the targets and descriptions from the comments next to them?

jmbowman replied:
Essentially.  Ned wrote a blog post about it (both versions): https://nedbatchelder.com/blog/201804/makefile_help_target.html
